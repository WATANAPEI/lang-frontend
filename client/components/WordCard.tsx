import React, { useState } from "react";
import styles from "../styles/style.css";
// @ts-ignore

export interface WordResponse {
  id: number;
  word: string;
  meaning: string;
  wordLanguageID: number;
  meaningLanguageID: number;
}

function WordCard(word: WordResponse) {
  const [isFlipped, setIsFlipped] = useState(false);
  return (
    <div className={styles.flipCard}>
      <div
        className={styles.flipCardInner}
        onClick={(event: React.MouseEvent<HTMLInputElement>) => {
          if (!isFlipped) {
            event.currentTarget.style.transform = "rotateY(180deg)";
            setIsFlipped(true);
          } else {
            event.currentTarget.style.transform = "rotateY(0deg)";
            setIsFlipped(false);
          }
        }}
      >
        <React.Fragment>
          <div className={styles.flipCardFront}>{word.word}</div>
          <div className={styles.flipCardBack}>{word.meaning}</div>
        </React.Fragment>
      </div>
    </div>
  );
}

export default WordCard;
