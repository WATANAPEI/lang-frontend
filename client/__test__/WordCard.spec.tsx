import React from "react";
import { mount, shallow } from "enzyme";
// @ts-ignore
import WordCard from "../components/WordCard.tsx";
// @ts-ignore
import { WordResponse } from "../components/WordCard.tsx";
import toJson from "enzyme-to-json";

describe("<WordCard />", () => {
  const testWord: WordResponse = {
    id: 1,
    word: "test word",
    meaning: "test meaning",
    wordLanguageID: 1,
    meaningLanguageID: 2
  };
  const wrapper = mount(<WordCard {...testWord} />);
  afterAll(() => {
    wrapper.unmount();
  })

  it("has 4 divs with flipCard/flipCardInner/flipCardFront/flipCardBack class", () => {
    expect(wrapper.find("div.flipCard")).toHaveLength(1);
    expect(wrapper.find("div.flipCardInner")).toHaveLength(1);
    expect(wrapper.find("div.flipCardFront")).toHaveLength(1);
    expect(wrapper.find("div.flipCardBack")).toHaveLength(1);
    //console.log(`wrapper: ${wrapper.debug()}`);
    expect(wrapper).toMatchSnapshot();
  });

  it("has a card with word", () => {
    expect(wrapper.find(".flipCardFront").text()).toEqual(testWord.word);
  });

  it("has a card with meaning", () => {
    expect(wrapper.find(".flipCardBack").text()).toEqual(testWord.meaning);
  });

  it("check the mount rendering doesn't change before and after click", () => {
    const wrapperString = wrapper.debug();
    wrapper.find(".flipCardInner").simulate("click");
    expect(wrapper.debug()).toEqual(wrapperString);
    wrapper.find(".flipCardInner").simulate("click");
    expect(wrapper.debug()).toEqual(wrapperString);
  });
});
