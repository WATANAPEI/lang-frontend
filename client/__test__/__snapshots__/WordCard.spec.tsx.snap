// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordCard /> has 4 divs with flipCard/flipCardInner/flipCardFront/flipCardBack class 1`] = `
<WordCard
  id={1}
  meaning="test meaning"
  meaningLanguageID={2}
  word="test word"
  wordLanguageID={1}
>
  <div
    className="flipCard"
  >
    <div
      className="flipCardInner"
      id="flipCardInner"
      onClick={[Function]}
    >
      <div
        className="flipCardFront"
      >
        test word
      </div>
      <div
        className="flipCardBack"
      >
        test meaning
      </div>
    </div>
  </div>
</WordCard>
`;
